app/grandchallenge/invoices/models.py

```diff
@@ -94,15 +98,90 @@ class Invoice(models.Model):
     class Meta:
         constraints = [
             models.CheckConstraint(
-                check=models.Q(payment_type__in=PaymentTypeChoices.values),
+                check=Q(payment_type__in=PaymentTypeChoices.values),
                 name="payment_type_in_choices",
             ),
             models.CheckConstraint(
-                check=models.Q(payment_status__in=PaymentStatusChoices.values),
+                check=Q(payment_status__in=PaymentStatusChoices.values),
                 name="payment_status_in_choices",
             ),
+            models.CheckConstraint(
+                name="issued_on_date_required_for_issued_payment_status",
+                check=Q(issued_on__isnull=False)
+                | ~Q(payment_status=PaymentStatusChoices.ISSUED)
+                | Q(payment_type=PaymentTypeChoices.COMPLIMENTARY),
+                violation_error_message="When setting the payment status to 'Issued',"
+                " you must set the 'Issued on' date.",
+            ),
+            models.CheckConstraint(
+                name="paid_on_date_required_for_paid_payment_status",
+                check=Q(paid_on__isnull=False)
+                | ~Q(payment_status=PaymentStatusChoices.PAID)
+                | Q(payment_type=PaymentTypeChoices.COMPLIMENTARY),
+                violation_error_message="When setting the payment status to 'Paid',"
+                " you must set the 'Paid on' date.",
+            ),
+            models.CheckConstraint(
+                name="comments_required_for_complimentary_payment_type",
+                check=Q(internal_comments__gt="")
+                | ~Q(payment_type=PaymentTypeChoices.COMPLIMENTARY),
```

Comment:
Maybe `check=~(Q(payment_type=PaymentTypeChoices.COMPLIMENTARY) & Q(internal_comments=""))` is more readable?